### ISSUE TITLE
Support additional Geocoding providers

### ISSUE THREAD

**Issue by jom** [Automattic/WP-Job-Manager#1753]
We should add additional providers for Geocoding data. 

Some possibilities:
- [Mapbox](https://docs.mapbox.com/api/search/)
- [Geocod.io](https://Geocod.io) (US / Canada only 😢)
- [locationiq.com](https://locationiq.com) (US / Canada only 😢)

I think to start off with, Mapbox makes the most sense as an _additional_ provider. We'd still default to Google Maps API for backwards compatibility, but would allow people to select an alternative provider with its own API key.

**Comment by richardmtl** [comment]
Also requested in https://wordpress.org/support/topic/map-choice-open-street-map/

**Comment by richardmtl** [comment]
Also interested: 2322493-zen

**Comment by Gnodesign** [comment]
Any news on this?

**Comment by jom** [comment]
@Gnodesign I have started on it, but won't make it in this release so it is bumped to next major (no ETA). It'll just be Mapbox at first.

**Comment by Gnodesign** [comment]
Hey Jom,

Thanks for getting back. That is great news, we definitely need a secondary Geocoding Provider and mapbox is a great pick. A lot of users are trying to avoid google maps, this will be a great feature for the plugin.

Will be looking forward for this. :)

**Comment by PabloCulebras** [comment]
> We should add additional providers for Geocoding data.
> 
> Some possibilities:
> 
> * [Mapbox](https://docs.mapbox.com/api/search/)
> * [Geocod.io](https://Geocod.io) (US / Canada only 😢)
> * [locationiq.com](https://locationiq.com) (US / Canada only 😢)
> 
> I think to start off with, Mapbox makes the most sense as an _additional_ provider. We'd still default to Google Maps API for backwards compatibility, but would allow people to select an alternative provider with its own API key.

Yes please, add Mabox as a provider! It's a long time since I am wishing a Mapbox-Wordpress easy integration... mapbox offers an easy way to design custom mapa, but unfortunately there seems not to be any UI to add layers, markers, categories, etc to it within Worpress! I think it would be extremely useful

**Comment by wpfan2099** [comment]
hi at all
Thanks for getting back. That is great news, we definitely need a secondary Geocoding Provider and mapbox is a great pick
and osm too:  https://wordpress.org/support/plugin/osm/

**Comment by tarifa10** [comment]
see an overview on more than twenty-osm-plugins that are avaialble:
https://community.apachefriends.org/f/viewtopic.php?f=2&t=79902

**Comment by cena** [comment]
https://wordpress.org/support/topic/support-additional-geocoding-providers-for-wp-job-manager-openstreetmap-support/

**Comment by Gnodesign** [comment]
@jom Any updates on this? I know things are a little bit slower due to the current situation but it would be great if we could get an update on how this is going. Additional Geocoding providers are really needed for the plugin especially since Google Maps switched to paid plans.

**Comment by gikaragia** [comment]
6717219-zd-a8c
